### DISCUSSION TITLE
Show a menu on Handle Selection

### DISCUSSION THREAD

**Discussion by emmatraversy** [xyflow/xyflow#3885] [Q&A] [answered]
Hi everyone.

I have custom nodes and some of them have multiple handles (For example conditional node has True or False handles).
I want to create a Add Menu similar to the image below (but on the handles) to show after clicking handles. So, then I use the current node and handle data to update the nodes and edges data and add the selected custom node to the data.
![image](https://github.com/xyflow/xyflow/assets/111432008/5bc144ca-500d-4bb8-bed9-9cc1d65febd7)
to clear my point: I want to add the new nodes by this menu not by clicking and dragging.

I've searched the issues and discussions but couldn't find the proper way or event.
I was wondering if there is a more standard way to do this and if there is a suitable handler in the framework.
I just need a clue to go on.

Many Thanks in advance
Regards
Emma

**Comment by moklick** [comment]
You can add nodes and edges programmatically by adding them to your nodes and edges state. You can use the helper from `useReactFlow` or directly update your store (in case you use one). Example: https://codesandbox.io/p/sandbox/unruffled-sound-hl6h56?file=/src/CustomNode.tsx:13,2-34,1

**Reply by emmatraversy** [reply]
Dear @moklick thank you for your quick reply
Unfortunately, I couldn't open the link (Sandbox not found)

The part of adding nodes and edges programmatically, I'm okay with it.
My problem is creating a menu that shows on custom Handle component when clicked(or hover). 
Is there a handler in the framework for Click on Handle Event or something like Node Toolbar example but for Handle?
Or I should create something with onclick event and open/close state like normal react apps?

Thank u so much.

**Reply by moklick** [reply, marked as answer]
This link should work: https://codesandbox.io/p/devbox/add-nodes-hl6h56?file=%2Fsrc%2FCustomNode.tsx%3A4%2C22

I would wrap the Handle component with a div and then add the click handler there.

**Reply by emmatraversy** [reply]
@moklick Thank you very much for your quick response.
Links works fine now.


> I would wrap the Handle component with a div and then add the click handler there.

I'm going to do it this way.

Thanks again!
